src/assets/wise5/classroomMonitor/dataExport/strategies/AbstractComponentDataExportStrategy.ts

```diff
@@ -0,0 +1,393 @@
+import { AbstractDataExportStrategy } from './AbstractDataExportStrategy';
+
+export abstract class AbstractComponentDataExportStrategy extends AbstractDataExportStrategy {
+  canViewStudentNames: boolean;
+  includeOnlySubmits: boolean;
+  includeStudentNames: boolean;
+  workSelectionType: string;
+
+  constructor(protected nodeId: string, protected component: any, additionalParams: any) {
+    super();
+    this.canViewStudentNames = additionalParams.canViewStudentNames;
+    this.includeOnlySubmits = additionalParams.includeOnlySubmits;
+    this.includeStudentNames = additionalParams.includeStudentNames;
+    this.workSelectionType = additionalParams.workSelectionType;
+  }
+
+  abstract generateComponentWorkRow(
+    component: any,
+    columnNames: string[],
+    columnNameToNumber: any,
+    rowCounter: number,
+    workgroupId: number,
+    userId1: number,
+    userId2: number,
+    userId3: number,
+    studentName1: string,
+    studentName2: string,
+    studentName3: string,
+    periodName: string,
+    componentRevisionCounter: any,
+    componentState: any
+  ): string[];
+
+  generateComponentHeaderRow(columnNames: string[], columnNameToNumber: any): string[] {
+    this.populateColumnNames(columnNames, columnNameToNumber);
+    const headerRow = [];
+    for (const columnName of columnNames) {
+      headerRow.push(columnName);
+    }
+    return headerRow;
+  }
+
+  private populateColumnNames(columnNames: string[], columnNameToNumber: any): void {
+    const defaultColumnNames = [
+      '#',
+      'Workgroup ID',
+      'User ID 1',
+      'Student Name 1',
+      'User ID 2',
+      'Student Name 2',
+      'User ID 3',
+      'Student Name 3',
+      'Class Period',
+      'Project ID',
+      'Project Name',
+      'Run ID',
+      'Start Date',
+      'End Date',
+      'Server Timestamp',
+      'Client Timestamp',
+      'Node ID',
+      'Component ID',
+      'Component Part Number',
+      'Step Title',
+      'Component Type',
+      'Component Prompt',
+      'Student Data',
+      'Component Revision Counter',
+      'Is Submit',
+      'Submit Count'
+    ];
+    for (let c = 0; c < defaultColumnNames.length; c++) {
+      const defaultColumnName = defaultColumnNames[c];
+      columnNameToNumber[defaultColumnName] = c;
+      columnNames.push(defaultColumnName);
+    }
+  }
+
+  generateComponentWorkRows(
+    component: any,
+    columnNames: string[],
+    columnNameToNumber: any,
+    nodeId: string
+  ): any[] {
```

Comment:
Looks like this returns a string array?
```suggestion
  ): string[] {
```